crates/matrix-sdk-sqlite/src/state_store.rs

```diff
@@ -594,20 +604,22 @@ trait SqliteObjectStateStoreExt: SqliteObjectExt {
         event_type: Key,
         state_keys: Vec<Key>,
     ) -> Result<Vec<(bool, Vec<u8>)>> {
-        let sql_params = vec!["?"; state_keys.len()].join(", ");
-        let sql = format!(
-            "SELECT stripped, data FROM state_event
+        chunk_large_query_over(state_keys, None, move |state_keys: Vec<Key>| {
+            let sql_params = repeat_vars(state_keys.len());
```

Comment:
Even more efficient:

```suggestion
            use itertools::Itertools;
            let sql_params = std::iter::repeat("?").take(state_keys.len()).format(", ");
```

Reply:
TIL, thanks!